### PR TITLE
Update Voltage Level creation form

### PR DESCRIPTION
**Pull request by ayolab**

### CODE REVIEW

**Review by FranckLecuyer — COMMENTED** [review]
Tests:
When reediting a voltage level creation form and resetting to empty the optional low voltage limit, high voltage limit, low short circuit limit and high circuit limit, an error occured on validation : "Field value format is incorrect"

File: src/components/refactor/dialogs/line-attach-to-voltage-level/line-attach-to-voltage-level-dialog.js [hunk 1]
```diff
@@ -118,6 +119,12 @@ const LineAttachToVoltageLevelDialog = ({
                 }),
             });
             setAttachmentLine(lineAttach?.attachmentLine);
+            const newVoltageLevel = lineAttach?.mayNewVoltageLevelInfos;
```
Comment (FranckLecuyer) [inline src/components/refactor/dialogs/line-attach-to-voltage-level/line-attach-to-voltage-level-dialog.js:122]: newVoltageLevel can be undefined, no ?

File: src/components/refactor/dialogs/line-attach-to-voltage-level/line-attach-to-voltage-level-dialog.js [hunk 2]
```diff
@@ -118,6 +119,12 @@ const LineAttachToVoltageLevelDialog = ({
...
+            newVoltageLevel.busbarSections = buildNewBusbarSections(
+                lineAttach?.mayNewVoltageLevelInfos?.equipmentId,
+                lineAttach?.mayNewVoltageLevelInfos?.sectionCount,
+                lineAttach?.mayNewVoltageLevelInfos?.busbarCount
+            );
             setNewVoltageLevel(lineAttach?.mayNewVoltageLevelInfos);
```
Comment (FranckLecuyer) [inline src/components/refactor/dialogs/line-attach-to-voltage-level/line-attach-to-voltage-level-dialog.js:128]: setNewVoltageLevel(newVoltageLevel); ?

File: src/components/refactor/dialogs/line-split-with-voltage-level/line-split-with-voltage-level-dialog.js [hunk 3]
```diff
@@ -103,7 +104,13 @@ const LineSplitWithVoltageLevelDialog = ({
                         lineSplit?.mayNewVoltageLevelInfos?.equipmentId,
                 }),
             });
-            setNewVoltageLevel(lineSplit?.mayNewVoltageLevelInfos);
+            const newVoltageLevel = lineSplit?.mayNewVoltageLevelInfos;
```
Comment (FranckLecuyer) [inline src/components/refactor/dialogs/line-split-with-voltage-level/line-split-with-voltage-level-dialog.js:107]: newVoltageLevel can be undefined, no ?

File: src/components/refactor/dialogs/line-to-attach-or-split-form/line-to-attach-or-split-utils.js [hunk 4]
```diff
@@ -41,3 +41,20 @@ export const getLineToAttachOrSplitFormData = ({
...
+    const newBusbarSections = [];
+    for (let i = 0; i < sectionCount; i++) {
+        for (let j = 0; j < busbarCount; j++) {
+            newBusbarSections.push({
+                id: equipmentId + '_' + (j + 1) + '_' + (i + 1),
+                name: '',
+            });
+        }
+    }
+    return newBusbarSections.sort((a, b) => a?.id?.localeCompare(b?.id));
```
Comment (FranckLecuyer) [inline src/components/refactor/dialogs/line-to-attach-or-split-form/line-to-attach-or-split-utils.js:59]: if you loop first on busbarCount and then on sectionCount, there is not need to sort, no ?

File: src/components/refactor/dialogs/voltage-level-creation/coupling-omnibus/coupling-omnibus-form.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    };
+
+    const voltageLevelID = useWatch({ name: EQUIPMENT_ID });
+    const busBarCount = useWatch({ name: BUS_BAR_COUNT });
+    const sectionCount = useWatch({ name: SECTION_COUNT });
+
+    const sectionOptions = useMemo(() => {
+        const options = [];
+        if (voltageLevelID && busBarCount && sectionCount) {
+            for (let i = 0; i < sectionCount; i++) {
```
Comment (FranckLecuyer) [inline src/components/refactor/dialogs/voltage-level-creation/coupling-omnibus/coupling-omnibus-form.js:35]: mutualize code with code in buildNewBusbarSections function ?

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/components/refactor/dialogs/voltage-level-creation/coupling-omnibus/coupling-omnibus-form.js (hunk 5 above)
Comment (ayolab) [reply src/components/refactor/dialogs/voltage-level-creation/coupling-omnibus/coupling-omnibus-form.js]: Done

File: src/translations/en.json [hunk 6]
```diff
@@ -566,9 +579,11 @@
     "VoltageLevelCreationError": "Voltage level creation error",
     "Breaker": "Breaker",
     "Disconnector": "Disconnector",
+    "BREAKER": "Breaker",
```
[the remark this answers is not in the document]
Comment (ayolab) [reply src/translations/en.json:582]: It is used to translate the values received from the backend while filling the "switchesBetweenSections" field

**Comment by ayolab** [comment]
> Tests: When re-editing a voltage level creation form and resetting to empty the optional low voltage limit, high voltage limit, low short circuit limit and high circuit limit, an error occured on validation : "Field value format is incorrect"

Fixed

**Review by FranckLecuyer — APPROVED** [review]

**Review by YenguiSeddik — APPROVED** [review]

**Pull request merged**